test/e2e/azure_logcollector.go

```diff
@@ -108,14 +109,19 @@ func (k AzureLogCollector) CollectMachinePoolLog(ctx context.Context, management
 	}
 
 	for i, instance := range mp.Spec.ProviderIDList {
-		hostname := mp.Status.NodeRefs[i].Name
+		if mp.Status.NodeRefs != nil && len(mp.Status.NodeRefs) >= (i+1) {
+			hostname := mp.Status.NodeRefs[i].Name
 
-		if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isWindows, filepath.Join(outputPath, hostname)); err != nil {
-			errors = append(errors, err)
-		}
+			if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isWindows, filepath.Join(outputPath, hostname)); err != nil {
+				errors = append(errors, err)
+			}
 
-		if err := collectVMSSBootLog(ctx, instance, filepath.Join(outputPath, hostname)); err != nil {
-			errors = append(errors, err)
+			if err := collectVMSSBootLog(ctx, instance, filepath.Join(outputPath, hostname)); err != nil {
+				errors = append(errors, err)
+			}
+		} else {
+			Logf(fmt.Sprintf("INFO: MachinePool instance %s does not have a corresponding NodeRef", instance))
```

Comment:
Logf is already printing "INFO:" so this would render as `INFO: INFO: `. Also it takes in format + args so you don't need to use `fmt.Sprintf`:

```suggestion
			Logf("MachinePool instance %s does not have a corresponding NodeRef", instance)
```

It seems that azure_logcollector.go is double printing INFO in a few places, eg. `Apr 20 23:09:07.082: INFO: INFO: Collecting boot logs for AzureMachine capz-e2e-002817-ipv6-md-0-vcpmh`, might be good to fix up those too while you're here.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/1703ef3434cdbe35a31062e1abb04e4291fdb9fc/test/e2e/log.go#L40

Reply:
good idea, done